### ISSUE TITLE
[GraphQL] Default access control

### ISSUE THREAD

**Issue by lukasluecke** [api-platform/core#2946]
Right now when you enable GraphQL every operation is allowed without any access checks. I think it would be great to have a simple configuration option that would call `is_granted` by default, so I don't have to write the exact same boilerplate in each and every resource.

**Comment by lukasluecke** [comment]
@alanpoulain This might also be a good candidate for the GraphQL resolver events you mentioned in #2944 - then we wouldn't need another configuration option somewhere, but could just add an example of a Listener to the documentation - or even include it and let users enable / load it.

**Comment by alanpoulain** [comment]
I think there is already this feature today by using the `access_control` attribute: https://api-platform.com/docs/core/graphql/#security-access_control
But speaking generally, yes it should be possible to extend features of the resolvers. Events could be a solution. But I think @dunglas prefers to have extension points. I need the @api-platform/core-team to tell me what they think.

**Comment by lukasluecke** [comment]
Sorry if it wasn't clear from my original post, but that is the "boilerplate" I was talking about. I don't want to have the following in each of my resources (and I might forget to add it to newly created resources) - that's why something like it as a "preset" would be nice 🙂

```
 * graphql={
 *      "create"={
 *          "access_control"="is_granted('create', object)",
 *      },
 *      "update"={
 *          "access_control"="is_granted('update', object)",
 *      },
 *      "delete"={
 *          "access_control"="is_granted('delete', object)",
 *      }
 * }
```

**Comment by alanpoulain** [comment]
Makes sense.

**Comment by dunglas** [comment]
If we add such way to globally configure defaults for resources, we should do it for REST too. It could also be nice to have for the (de)normalization context, and other attributes. So my proposal is to add a new configuration entry allowing to set default attributes that will automatically be applied to all resources.

It can be done by introducing a new `ResourceMetadataFactory` class.

**Comment by lukasluecke** [comment]
@dunglas Sounds good! I will see if I can find the time to work on a PR soon, but can't promise anything 🙈🙂

**Comment by lukasluecke** [comment]
Will be solved with #3099.
